src/main/java/com/huawei/cloud/servicestage/client/ServiceStageClient.java

```diff
@@ -182,7 +185,10 @@ public SimpleResponse updateService(String instanceId, String requestBody,
         request.setHeader(X_AUTH_TOKEN_HEADER_KEY, token.getToken());
 
         // proxy (if needed)
-        Util.setProxy(request);
+        if (!requestUrl.contains("localhost")) {

```

Comment:
there are too many repeating of the following lines in the code, could it be added to the Util.setProxy() method instead? 

if (!requestUrl.contains("localhost")) {
   Util.setProxy(request);
}

Reply:
@paulmchen code is updated and logic pushed to the Util class.